src/tests/ftest/launch.py

```diff
@@ -953,6 +958,161 @@ def rename_logs(avocado_logs_dir, test_file):
                 test_logs_dir, new_test_logs_dir, error))
 
 
+USE_DEBUGINFO_INSTALL = True
+
+def install_debuginfos():
+    """Install debuginfo packages"""
+
+    install_pkgs = [{'name': 'gdb'}, {'name': 'python-magic'}]
+    cmds = []
+
+    if USE_DEBUGINFO_INSTALL:
+        cmds.append("sudo debuginfo-install -y "                   \
+                    "--exclude ompi-debuginfo,gcc-debuginfo,"      \
+                               "gcc-base-debuginfo "               \
+                    "daos-server cart libpmemobj python openmpi3")
+    else:
+        import yum
+
+        yum_base = yum.YumBase()
+        yum_base.conf.assumeyes = True
+        yum_base.setCacheDir(force=True, reuse=True)
+        yum_base.repos.enableRepo('*debug*')
+
+        debuginfo_map = {'glibc':   'glibc-debuginfo-common',
+                         'libpmem': 'pmdk-debuginfo'}
+
+        # We're not using the yum API to install packages
+        # See the comments below.
+        #kwarg = {'name': 'gdb'}
+        #yum_base.install(**kwarg)
+
+        for pkg in ['python', 'glibc', 'daos', 'systemd', 'ndctl', 'libpmem',
+                    'mercury', 'cart', 'libfabric', 'argobots']:
+            try:
+                debug_pkg = debuginfo_map[pkg]
+            except KeyError:
+                debug_pkg = pkg + "-debuginfo"
+            try:
+                pkg_data = yum_base.rpmdb.returnNewestByName(name=pkg)[0]
+            except yum.Errors.PackageSackError as expn:
+                if expn.__str__().rstrip() == "No Package Matching " + pkg:
+                    print("Package {} not installed, "
+                          "skipping debuginfo".format(pkg))
+                    continue
+                else:
+                    raise
+            # This is how you actually use the API to add a package
+            # But since we need sudo to do it, we need to call out to yum
+            #kwarg = {'name': debug_pkg,
+            #         'version': pkg_data['version'],
+            #         'release': pkg_data['release']}
+            #yum_base.install(**kwarg)
+            install_pkgs.append({'name': debug_pkg,
+                                 'version': pkg_data['version'],
+                                 'release': pkg_data['release'],
+                                 'epoch': pkg_data['epoch']})
+
+    # This is how you normally finish up a yum transaction, but
+    # again, we need to employ sudo
```

Comment:
Could the https://pypi.org/project/elevate/ module solve the problem?

Reply:
Perhaps, in a future iteration.  This task has already taken much longer than it should have.